app/actions/remote/systems.ts

```diff
@@ -16,36 +15,89 @@ export type ConfigAndLicenseRequest = {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    try {
+        const {data: globalPolicy, error: globalPolicyError} = await fetchGlobalDataRetentionPolicy(serverUrl);
+        const {data: teamPolicies, error: teamPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl);
+        const {data: channelPolicies, error: channelPoliciesError} = await fetchAllGranularDataRetentionPolicies(serverUrl, true);
+
+        const hasError = globalPolicyError || teamPoliciesError || channelPoliciesError;
+        if (hasError) {
+            return hasError;
+        }
+
+        return {
+            globalPolicy,
+            teamPolicies: teamPolicies as TeamDataRetentionPolicy[],
+            channelPolicies: channelPolicies as ChannelDataRetentionPolicy[],
+        };
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+};
+
+export const fetchGlobalDataRetentionPolicy = async (serverUrl: string): Promise<{data?: GlobalDataRetentionPolicy; error?: unknown}> => {
     let client;
     try {
         client = NetworkManager.getClient(serverUrl);
     } catch (error) {
         return {error};
     }
 
-    let data = {};
     try {
-        data = await client.getDataRetentionPolicy();
+        const data = await client.getGlobalDataRetentionPolicy();
+        return {data};
     } catch (error) {
         forceLogoutIfNecessary(serverUrl, error as ClientError);
         return {error};
     }
+};
+
+export const fetchAllGranularDataRetentionPolicies = async (
+    serverUrl: string,
+    isChannel = false,
+    page = 0,
+    policies: Array<TeamDataRetentionPolicy | ChannelDataRetentionPolicy> = [],
+): Promise<{data?: Array<TeamDataRetentionPolicy | ChannelDataRetentionPolicy>; error?: unknown}> => {
+    let client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
 
     const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-    if (operator) {
-        const systems: IdValue[] = [{
-            id: SYSTEM_IDENTIFIERS.DATA_RETENTION_POLICIES,
-            value: JSON.stringify(data),
-        }];
-
-        operator.handleSystem({systems, prepareRecordsOnly: false}).
-            catch((error) => {
-                logError('An error occurred while saving data retention policies', error);
-            });
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
     }
 
-    return data;
+    const {database} = operator;
+
+    const currentUserId = await getCurrentUserId(database);
+    let data;
+    if (isChannel) {
+        data = await client.getChannelDataRetentionPolicies(currentUserId, page);
+    } else {
+        data = await client.getTeamDataRetentionPolicies(currentUserId, page);
+    }
+    policies.push(...data.policies);
```

Comment:
This is a slightly weird behaviour, since you are modifying an argument of the function. Is this intended? If not, you may want to spread also the argument (policies).

Reply:
still super weird and error prone